### PR TITLE
docs: add dev page for temporary settings

### PR DESCRIPTION
**Pull request by limitedmage**
Adds a docs page explaining what temporary settings are, what is the difference between temporary settings and site settings, and how to use temporary settings.

### CODE REVIEW

**Comment by sqs** [comment]
Just wanted to say thanks for writing this. I had come across temporary settings in the code and didn't know what they were for, and this page explains it very well.

**Comment by limitedmage** [comment]
> Just wanted to say thanks for writing this. I had come across temporary settings in the code and didn't know what they were for, and this page explains it very well.

Thank you! This is a brand new infrastructure feature and I'm hoping good docs will help it be widely adopted (and we can more all our "popup/tour-dismissed" settings currently in localStorage to this).

File: doc/dev/background-information/web/temporary_settings.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+Basic user settings that should be retained for users across sessions 
+can be stored as temporary settings. These should be trivial settings
+that would be fine if they were lost.
+
+For authenticated users, temporary settings are stored in the database 
+in the `temporary_setings` table and are queried and modified via the GraphQL API.
+
+For unauthenticated users, temporary settings are stored in `localStorage`. 
+
+## Different between temporary settings and site settings
```
Comment (valerybugakov) [inline doc/dev/background-information/web/temporary_settings.md:12]: Not a native speaker here 🤗

```suggestion
## Difference between temporary settings and site settings
```

File: doc/dev/background-information/web/temporary_settings.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+Update the interface [`TemporarySettingsSchema` in `TemporarySettings.ts`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/web/src/settings/temporary/TemporarySettings.ts?L8:18) 
+by adding a key for the setting you want to store. The key should be namespaced based on 
+the area of the site that will be using the settings. Example names include `'search.collapsedSidebarSections'` 
+or `'codeInsights.hiddenCharts'`. The value of the setting can be any JSON-serializable type.
+
+### Getting and setting settings
+
+Use the React hook [`useTemporarySetting`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/web/src/settings/temporary/useTemporarySetting.ts?L14:33) 
+to get an up-to-date value of the setting and a function that can update the value.
```
Comment (valerybugakov) [inline doc/dev/background-information/web/temporary_settings.md:62]: It would be great to inline a usage example here.

**Review by valerybugakov — APPROVED** [review]
That's awesome! Thanks for documenting it.

File: doc/dev/background-information/web/temporary_settings.md (hunk 1 above)
Comment (limitedmage) [reply doc/dev/background-information/web/temporary_settings.md:12]: I need Grammarly in VS Code 😅

Comment (limitedmage) [reply doc/dev/background-information/web/temporary_settings.md]: it exists :O https://marketplace.visualstudio.com/items?itemName=znck.grammarly

File: doc/dev/background-information/web/temporary_settings.md (hunk 2 above)
Comment (limitedmage) [reply doc/dev/background-information/web/temporary_settings.md]: Added, thanks for the suggestion!

**Pull request merged**